Add v1.0.0 migration document

adrienthebo opened the pull request.

aaron-lane commented on docs/MIGRATING_V1.md:
```diff
@@ -0,0 +1,316 @@
+# Upgrading to Project Factory v1.0
```
Per our style guide, the file name and the title should match. I think the file name can be more verbose.

adrienthebo replied:
Renamed to `docs/upgrading_to_project_factory_v1.0.md`

aaron-lane commented on docs/MIGRATING_V1.md:
```diff
@@ -0,0 +1,316 @@
+  credentials = "${file(var.credentials_path)}"
+}
+
+provider "gsuite" {
```
Is the G Suite provider necessary when the integration is disabled?

adrienthebo replied:
Yes, the gsuite provider produces errors otherwise. If you'd like to verify this you can check against the example code in this document.

aaron-lane commented on docs/MIGRATING_V1.md:
```diff
@@ -0,0 +1,316 @@
+managed. A state migration script is provided to update your existing states
+to minimize or eliminate resource re-creations. 
+
+NOTE: This document is a work in progress as v1.0 has not yet been released.
```
Can this be dropped assuming we just wait for the 1.0 release to merge this?

adrienthebo replied:
Yes, that note was specifically for this pull request review and not the final document.

adrienthebo replied:
Comment dropped.

aaron-lane commented on docs/MIGRATING_V1.md:
````diff
@@ -0,0 +1,316 @@
+#### 3. Download the state migration script
+
+```
+curl -O https://raw.githubusercontent.com/terraform-google-modules/terraform-google-project-factory/adrienthebo/migration-script/helpers/migrate.py
````
This should reference the v1.0 tag.

aaron-lane commented on docs/MIGRATING_V1.md:
```diff
@@ -0,0 +1,316 @@
+
+ module "project-factory" {
+-  source            = "github.com/terraform-google-modules/terraform-google-project-factory?ref=51665dd"
++  source            = "github.com/terraform-google-modules/terraform-google-project-factory?ref=8bdcac1"
```
This should reference the v1.0 tag.

aaron-lane commented on docs/MIGRATING_V1.md:
```diff
@@ -0,0 +1,316 @@
+}
+
+module "project-factory" {
+  source            = "github.com/terraform-google-modules/terraform-google-project-factory?ref=5791b7680037ef1bf64045e4ac0039f33a7808e6"
```
This should reference the v0.2.1 tag or the v0.2.2 tag.

aaron-lane commented on docs/MIGRATING_V1.md:
```diff
@@ -0,0 +1,316 @@
+}
+
+module "project-factory" {
+  source            = "github.com/terraform-google-modules/terraform-google-project-factory?ref=51665dd"
```
This should reference the v0.2.1 tag or the v0.2.2 tag.

aaron-lane commented on docs/MIGRATING_V1.md:
```diff
@@ -0,0 +1,316 @@
+ 
+ module "project-factory" {
+-  source             = "github.com/terraform-google-modules/terraform-google-project-factory?ref=51665dd"
++  source             = "github.com/terraform-google-modules/terraform-google-project-factory//modules/gsuite_enabled?ref=8bdcac1"
```
This should reference the v1.0 tag.

aaron-lane commented on docs/MIGRATING_V1.md:
````diff
@@ -0,0 +1,316 @@
+#### 3. Download the state migration script
+
+```
+curl -O https://raw.githubusercontent.com/terraform-google-modules/terraform-google-project-factory/adrienthebo/migration-script/helpers/migrate.py
````
This should reference the v1.0 tag.

adrienthebo commented:
Regarding all of the versions referenced in this pull request, I'm aware that they're pointing at the wrong commits - but we need the v0.2.2 bugfix to ensure the migration produces as few changes as possible, v1.0.0 isn't released, and the migration script hasn't been merged. All of the version or branch values should be considered placeholders until the required branches are merged and the versions are released.

adrienthebo commented:
I've updated the starting source and version to the v0.3.0 Terraform registry module, and updated the instructions to use v1.0.0 from the registry as well. The migration script will be fetched from the v1.0.0 tag of the source on GitHub.

adrienthebo commented:
@aaron-lane could you give this PR another review?

adrienthebo commented:
@morgante your changes LGTM. Since this is my PR I can't :+1: my own PR, so consider this a soft :+1:.
